### PR TITLE
New package ADIOS2

### PR DESCRIPTION
**Pull request by eschnett**
[ADIOS2](https://github.com/ornladios/ADIOS2) is a high-performance I/O library, comparable to HDF5.

### CODE REVIEW

File: A/ADIOS2/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd ADIOS2-2.7.1
+mkdir build
+cd build
+if [[ "$target" == x86_64-w64-mingw32 ]]; then
+    mpiopts="-DMPI_HOME=$WORKSPACE/destdir -DMPI_GUESS_LIBRARY_NAME=MSMPI -DMPI_C_LIBRARIES=msmpi64 -DMPI_CXX_LIBRARIES=msmpi64"
+elif [[ "$target" == *-mingw* ]]; then
+    mpiopts="-DMPI_HOME=$WORKSPACE/destdir -DMPI_GUESS_LIBRARY_NAME=MSMPI"
```
Comment (giordano) [inline A/ADIOS2/build_tarballs.jl:22]:
```suggestion
    mpiopts="-DMPI_HOME=${prefix} -DMPI_GUESS_LIBRARY_NAME=MSMPI -DMPI_C_LIBRARIES=msmpi64 -DMPI_CXX_LIBRARIES=msmpi64"
elif [[ "$target" == *-mingw* ]]; then
    mpiopts="-DMPI_HOME=${prefix} -DMPI_GUESS_LIBRARY_NAME=MSMPI"
```

File: A/ADIOS2/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+else
+    mpiopts=
+fi
+cmake -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN%.*}_gcc.cmake -DCMAKE_BUILD_TYPE=Release -DADIOS2_USE_Fortran=OFF -DADIOS2_BUILD_EXAMPLES=OFF -DBUILD_TESTING=OFF ${mpiopts} ..
+make -j$(nproc)
+make -j$(nproc) install
```
Comment (giordano) [inline A/ADIOS2/build_tarballs.jl:28]:
```suggestion
make -j${nproc}
make -j${nproc} install
```

File: A/ADIOS2/build_tarballs.jl (hunk 1 above)
Comment (eschnett) [reply A/ADIOS2/build_tarballs.jl]: Done.

File: A/ADIOS2/build_tarballs.jl (hunk 2 above)
Comment (eschnett) [reply A/ADIOS2/build_tarballs.jl]: Done.

File: A/ADIOS2/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+products = [
+    LibraryProduct("libadios2_c_mpi", :libadios2_c_mpi),
+    LibraryProduct("libadios2_c", :libadios2_c),
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae")),
+    Dependency(PackageSpec(name="MPICH_jll")),
+    Dependency(PackageSpec(name="MicrosoftMPI_jll")),
```
Comment (giordano) [inline A/ADIOS2/build_tarballs.jl:80]: If this works only for linux, this isn't needed, right?

Comment (eschnett) [reply A/ADIOS2/build_tarballs.jl:80]: According to the package documentation, other OSes should be supported, not just Linux. I have an issue open to clarify this. But yes – at the moment, this is not needed.

**Comment by eschnett** [comment]
It's good to go; Windows and Free BSD are not important to me.

**Pull request merged**